pontoon/base/models.py

```diff
@@ -25,12 +25,101 @@
 log = logging.getLogger('pontoon')
 
 
+class UserTranslationsManager(models.Manager):
+    """
+    Provides various method to interact with larger sets of translations and their stats for user.
+    """
+
+    def _changed_translations_counts(self, start_date=None, end_date=None, filter=None, exclude=None):
+        """
+        Helper method, returns expression object which allows us to annotate querysets
+        with counts of translations.
+        Method supports exclusion of fields, you can exclude field if param contains
+        exclude__ prefix
+        e.g.
+        _changed_translations_counts(exclude__approved=True)
+        Will exclude "approved" field with True value.
+        """
+        translation_query = (
+            ~Q(translation__string=F('translation__entity__string'))
+            & ~Q(translation__string=F('translation__entity__string'))
+        )
+
+        if filter:
+            translation_query &= Q(**filter)
+
+        if exclude:
+            translation_query &= ~Q(**exclude)
+
+        if start_date and end_date:
+            translation_query &= Q(translation__date__range=(start_date, end_date))
+
+        # For each translation that matches the filter, return 1. Aggregate
+        # the sum of all those results to count the number of matching
+        # translations.
+        return Sum(
+                Case(
```

Comment:
This is nitpicking, but the indentation in this block is really inconsistent. Probably stick to 4-space indents unless a hanging indent lined up with something would work better.